subprojects/gradle/build-verdict/src/main/kotlin/com/avito/android/build_verdict/BuildVerdictPlugin.kt

```diff
@@ -0,0 +1,36 @@
+package com.avito.android.build_verdict
+
+import com.avito.kotlin.dsl.isRoot
+import com.avito.utils.logging.ciLogger
+import com.google.gson.GsonBuilder
+import org.gradle.api.Plugin
+import org.gradle.api.internal.project.ProjectInternal
+import org.gradle.kotlin.dsl.create
+
+internal data class TaskPath(val value: String)
+
+class BuildVerdictPlugin : Plugin<ProjectInternal> {
+
+    override fun apply(target: ProjectInternal) {
+        require(target.isRoot()) {
+            "build-verdict plugin must be applied to the root project"
+        }
+        val logs = mutableMapOf<TaskPath, StringBuilder>()
```

Comment:
i'd wrap it to some `LogCapturer`, with interface, to easily change implementation, if this data structure show itself uncapable

Reply:
I don't see problems with changing implementation for internal field
But I could wrap this for you

Reply:
yep, it's not a problem in that perspective;

it's about design:
you pass implementation details around (mutable map), but imagine someone enabled `debug` logging level and we'll maybe encounter OOM's here while parsing logs, decided to change it to some buffered streaming;
i'd prefer to make an abstraction right away. backed by mutable map in naive implementation

and please, nothing here is "for me", i just sharing how i can improve this code :)

Reply:
> but imagine someone enabled debug logging level
We already capture only error level outputs, so we don't care about logging level

> nothing here is "for me"
I want to clarify my opinion that is cheaper to make "for you" then you could change my opinion in this design decision

Reply:
I thought and didn't get how LogCapturer would differ from Map
It would need read-write methods like a MutableMap.
The current code already segregates read and write operation. So I don't see how LogCapturer will improve the code